app/models/template_kind.rb

```diff
@@ -6,4 +6,25 @@ class TemplateKind < ActiveRecord::Base
   has_many :os_default_templates
   validates :name, :presence => true, :uniqueness => true
   scoped_search :on => :name
+
+  def self.jar
+    @jar ||= { "PXELinux" => N_("PXE Linux"),
+               "PXEGrub" => N_("PXE Grub"),
+               "iPXE" => N_("iPXE"),
+               "provision" => N_("Provision"),
+               "finish" => N_("Finish"),
+               "script" => N_("Script"),
+               "user_data" => N_("User Data"),
+               "ZTP" => N_("ZTP"),
+               "POAP" => N_("POAP")
+             }
+  end
+
+  def self.add_to_jar(hash)
+    jar.merge!(hash) { |key| raise Foreman::Exception.new(N_("Cannot add template with key %s, it already exists"), key) }
+  end
+
+  def to_label
```

Comment:
If you change this to `to_s` you can get rid of all the `.try(:to_label)` later since `nil.to_s => ""` and `to_s` is called implicitly when trying to print the object.